File: python/collatz-conjecture/collatz_conjecture.py
def steps(n: int, steps=0):
    if type(n) is not int: raise TypeError("Only int value types are allowed")
    if n < 1: raise ValueError("Only positive integers are allowed")

    while n != 1: n = n & 1 and 3*n + 1 or n >> 1; steps += 1
    return steps

File: javascript/collatz-conjecture/collatz-conjecture.js
// @ts-check

/** @param {number} n a positive integer */
export function steps(n, count=0) {
  if ((n |= 0) < 1) throw Error('Only positive numbers are allowed');
  while (n != 1) n = n & 1 && 3*n + 1 || n >> 1, ++count; return count; }
